samples/matter/lock/src/app_task.cpp

```diff
@@ -274,6 +288,71 @@ void AppTask::LockActionEventHandler(const AppEvent &event)
 	}
 }
 
+#ifdef CONFIG_THREAD_WIFI_SWITCHING
+void AppTask::SwitchModesEventHandler(const AppEvent &event)
+{
+	/* Swap the application from using Thread to Wi-Fi or opposite. */
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+	LOG_INF("Switching application from Thread to Wi-FI triggered.");
+	SystemLayer().ScheduleLambda([] {
+		SoftwareImagesSwapper swapper;
+		swapper.Swap(SoftwareImagesSwapper::ApplicationImageId::Image_2);
+	});
+#elif defined(CONFIG_CHIP_WIFI)
+	LOG_INF("Switching application from Wi-Fi to Thread triggered.");
+	SystemLayer().ScheduleLambda([] {
+		SoftwareImagesSwapper swapper;
+		swapper.Swap(SoftwareImagesSwapper::ApplicationImageId::Image_1);
+	});
+#endif
+}
+
+void AppTask::SwitchModesTimerTimeoutCallback(k_timer *timer)
+{
+	if (!timer) {
+		return;
+	}
+
+	Instance().mSwitchModesTimerActive = false;
```

Comment:
Theoretically if we don't want to support factory reset being possible to trigger while invoking the image swap we should check this flag in function button handler. Other than that maybe we do not need this flag at all (it's all handled in separate timer and correlated handler methods).

Reply:
I don't think it's possible to do the factory reset during DFU. If the DFU event will be fetched from the event queue, it will block fetching other application events until completing. This flag is useful only to decide whether releasing the button was due to cancellation of triggering swap operation or due to 10 s timeout expiration and starting swap. It's a little bit trivial reason, but maybe it's still worth to print such information.

Reply:
Well, currently a race is possible because you run the swapper on the CHIP thread, so all the buttons will still work (until you try to do something on a CHIP thread). But if you run the swapper on the main thread it should be safe, I guess.